include/mrdocs/Metadata/Javadoc.hpp

```diff
@@ -172,7 +242,7 @@ struct Text : Node
     bool equals(const Node& other) const noexcept override
     {
         return kind == other.kind &&
-            *this == static_cast<const Text&>(other);
+            *this == dynamic_cast<const Text&>(other);
```

Comment:
Why did you change these to `dynamic_cast`? `kind == other.kind` implies that `other` is derived from `Text`.

Reply:
It's a warning I get.

> Do not use static_cast to downcast from a base to a derived class; use dynamic_cast instead

https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/pro-type-static-cast-downcast.html